test/tryGetDefaultBranch.test.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import assert = require('assert');
+import { Uri } from 'vscode';
+import { cpUtils, getGitWorkspaceState, GitWorkspaceState, tryGetDefaultBranch } from '../extension.bundle';
+import { createTestActionContext, testFolderPath } from './global.test';
+
+suite('Get default branch for Git repo', function (this: Mocha.Suite): void {
+    this.timeout(30 * 1000);
+    const context = createTestActionContext();
+    const testFolderUri: Uri = Uri.file(testFolderPath);
+    const localDefaultBranch: string = 'localDefault';
+    const globalDefaultBranch: string = 'globalDefault';
+
+    test('Workspace with default branch set in global config', async () => {
+        await cpUtils.executeCommand(undefined, undefined, 'git', 'config', '--global', 'init.defaultBranch', globalDefaultBranch);
+        const gitWorkspaceState: GitWorkspaceState = await getGitWorkspaceState(context, testFolderUri);
+
+        if (!gitWorkspaceState.repo) {
+            throw new Error('Could not retrieve git repository.');
+        }
+
+        await gitWorkspaceState.repo.createBranch(globalDefaultBranch, false);
+        assert.strictEqual(await tryGetDefaultBranch(gitWorkspaceState.repo), globalDefaultBranch);
+    });
+
+    test('Workspace with default branch set in local config', async () => {
+        await cpUtils.executeCommand(undefined, testFolderPath, 'git', 'config', '--local', 'init.defaultBranch', localDefaultBranch);
+        const gitWorkspaceState: GitWorkspaceState = await getGitWorkspaceState(context, testFolderUri);
+
+        if (!gitWorkspaceState.repo) {
+            throw new Error('Could not retrieve git repository.');
+        }
+
+        await gitWorkspaceState.repo.createBranch(localDefaultBranch, false);
+        assert.strictEqual(await tryGetDefaultBranch(gitWorkspaceState.repo), localDefaultBranch);
+    });
+
+    suiteTeardown(async () => {
+        // reset the configs
+        await cpUtils.executeCommand(undefined, testFolderPath, 'git', 'config', '--local', 'init.defaultBranch', '');
```

Comment:
If you're going to "reset the configs", it should be back to what they originally had set, which might not be empty. But even then I'm just not sure we want to be messing with people's global git config like this because suiteTeardown doesn't always happen (for example if you stop your debugging session during mid tests). My advice would be to skip the global config and only do local config (because that only applies to the random test folder, right?)